src/arc/mp/MultiplayerPanel.as

```diff
@@ -359,14 +363,22 @@ package arc.mp
         private function nameRoom(room:Room):String
         {
             const level:int = room.level;
-            const color:int = ArcGlobals.getDivisionColor(level);
-            const title:String = ArcGlobals.getDivisionTitle(level);
-
-            const dulledColour:String = MultiplayerChat.textDullColour(color, 1).toString(16);
-            const roomName:String = "(" + title + ")";
             const spectatorString:String = (room.specCount > 0) ? "+" + room.specCount + " " : "";
 
-            return MultiplayerChat.textFormatSize(room.userCount + "/2 " + spectatorString, "-1") + MultiplayerChat.textFormatColour(MultiplayerChat.textEscape((room.isPrivate ? "!" : "") + roomName), "#" + dulledColour) + " " + room.name;
+            if (room.playerCount > 0 && level != -1)
+            {
+                const color:int = ArcGlobals.getDivisionColor(level);
+                const titleString:String = ArcGlobals.getDivisionTitle(level);
+
+                const dulledColour:String = MultiplayerChat.textDullColour(color, 1).toString(16);
+                const titlePrefix:String = "(" + titleString + ")";
+
+                return MultiplayerChat.textFormatSize(room.userCount + "/2 " + spectatorString, "-1") + MultiplayerChat.textFormatColour(MultiplayerChat.textEscape((room.isPrivate ? "!" : "") + titlePrefix), "#" + dulledColour) + " " + room.name;
+            }
+            else
+            {
+                return MultiplayerChat.textFormatSize(room.userCount + "/2 " + spectatorString, "-1") + " " + MultiplayerChat.textEscape((room.isPrivate ? "!" : "") + room.name);
```

Comment:
The `MultiplayerChat.textFormatSize(room.userCount + "/2 " + spectatorString, "-1")` part should be factored out of the if/else.